Sources/RedisKit/RedisSet.swift

```diff
@@ -1,143 +1,173 @@
-#warning("Re-implement RedisSet")
-//extension RedisDatabase {
-    /// Creates a `RedisSet` for the provided key, that contains the declared type of elements.
+extension RedisClient {
+    /// Creates a `RedisSet` reference to the value stored at `key` with values of `type`.
     ///
-    ///     let idSet = redis.createSetReference(fromKey: "ids", ofType: Int.self)
-    ///     // idSet represents a Set of `Int`.
+    ///     let setOfIDs = client.makeSetReference(key: String, type: Int.self)
+    ///     // setOfIDs represents a Set of `Int`.
     ///
-    /// - Parameter fromKey: The key to identify the Set in this `RedisClient`.
-    /// - Parameter ofType: The type of the elements contained within the set.
-//    public func createSetReference<T>(fromKey key: String, ofType type: T.Type) -> RedisSet<T> {
-//        return RedisSet(identifier: key, using: self)
-//    }
+    /// - Parameters:
+    ///     - key: The Redis key to identify the set.
+    ///     - type: The Swift type representation of the elements in the set.
+    /// - Returns: A `RedisSet` for the key and element type specified.
+    @inlinable
+    public func makeSetReference<T>(key: String, type: T.Type = T.self) -> RedisSet<T> {
+        return RedisSet(identifier: key, client: self)
+    }
 
-    /// Creates a `RedisSet` for the provided key, representing a `Collection` type.
+    /// Creates a `RedisSet` reference to the value stored at `key` with values of `type`.
     ///
-    ///     let idSet = redis.createSetReference(fromKey: "ids", ofType: [Int].self)
-    ///     // idSet represents a Set of `Int`.
+    ///     let setOfIDs = client.makeSetReference(key: String, type: [Int].self)
+    ///     // setOfIDs represents a Set of `Int`.
     ///
-    /// - Parameter fromKey: The key to identify the Set in this `RedisClient`.
-    /// - Parameter ofType: The `Collection` type this set represents.
-//    public func createSetReference<C: Collection>(fromKey key: String, ofType type: C.Type) -> RedisSet<C.Element>
-//        where C.Element: RESPValueConvertible
-//    {
-//        return RedisSet(identifier: key, using: self)
-//    }
-//}
+    /// - Parameters:
+    ///     - key: The Redis key to identify the set.
+    ///     - type: The Swift type representation of the elements in the set.
+    /// - Returns: A `RedisSet` for the key and element type specified.
+    @inlinable
+    public func makeSetReference<C: Collection>(
```

Comment:
I prefer to have `where C: Collection` at the end. The `where` syntax is more powerful, making it required in some places. So having consistency everywhere is nice.

```swift
@inlinable
public func makeSetReference<C>(key: String, type: C.Type = C.self) -> RedisSet<C.Element> 
    where C: Collection
{
    return RedisSet(identifier: key, client: self)
}
```

Reply:
I also think this method could use a different signature, to indicate it's for a collection. Maybe `makeSetReference(key:collection:)`. Even then though, I don't see why you wouldn't just use the other method that takes `type:` and use the collection's element type.

Reply:
I'm thinking of nuking this until someone asks for it - I had it at one point and I don't remember why anymore.